### ISSUE TITLE
Exception type of fields() on non-dataclass

### ISSUE THREAD

**Issue by Dutcho** [ericvsmith/dataclasses#57]
As it was decided (issue #8) that `fields()`, `astuple()`, `asdict()` will be module-level functions (_not_ member functions), users _can_ (but _shouldn’t_) call these for non-dataclasses too. I surmise this should raise `TypeError` (not `AttributeError`, which is raised when `fields()` accesses the `_MARKER` attribute now) to best inform users of their erroneous use
```
>>> import dataclasses
>>> dataclasses.astuple((1, 2, 3))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "...dataclasses.py", line
    return tuple(getattr(obj, name) for name in fields(obj))
  File "...dataclasses.py", line
    return getattr(cls, _MARKER)
AttributeError: 'tuple' object has no attribute '__dataclass_fields__'
```

**Comment by ilevkivskyi** [comment]
I also think it should be a ``TypeError`` with a clearer message.

**Comment by ericvsmith** [comment]
This is waiting on PR #52, I'll look at it after that's resolved.

**Comment by ericvsmith** [comment]
Fixed in **f6287018a09ac03623cd2878fcebd7b43c23e7e0**.
